corehq/messaging/scheduling/scheduling_partitioned/tests/test_dbaccessors_partitioned.py

```diff
@@ -0,0 +1,302 @@
+        self.assertIn(self.db1, settings.PARTITION_DATABASE_CONFIG['shards'])
+        self.assertIn(self.db2, settings.PARTITION_DATABASE_CONFIG['shards'])
+        self.assertEqual(settings.PARTITION_DATABASE_CONFIG['shards'][self.db1], [0, 1])
+        self.assertEqual(settings.PARTITION_DATABASE_CONFIG['shards'][self.db2], [2, 3])
+        self.assertEqual(set(partition_config.get_form_processing_dbs()), set([self.db1, self.db2]))
+        self.assertEqual(ShardAccessor.get_database_for_doc(self.p1_uuid), self.db1)
+        self.assertEqual(ShardAccessor.get_database_for_doc(self.p2_uuid), self.db2)
+
+    def test_save_alert_schedule_instance(self):
+        self.assertEqual(AlertScheduleInstance.objects.using(partition_config.get_proxy_db()).count(), 0)
+        self.assertEqual(AlertScheduleInstance.objects.using(self.db1).count(), 0)
+        self.assertEqual(AlertScheduleInstance.objects.using(self.db2).count(), 0)
+
+        instance = self.make_alert_schedule_instance(self.p1_uuid)
+        save_alert_schedule_instance(instance)
+
+        self.assertEqual(AlertScheduleInstance.objects.using(partition_config.get_proxy_db()).count(), 0)
+        self.assertEqual(AlertScheduleInstance.objects.using(self.db1).count(), 1)
+        self.assertEqual(AlertScheduleInstance.objects.using(self.db2).count(), 0)
+
+    def test_save_timed_schedule_instance(self):
+        self.assertEqual(TimedScheduleInstance.objects.using(partition_config.get_proxy_db()).count(), 0)
+        self.assertEqual(TimedScheduleInstance.objects.using(self.db1).count(), 0)
+        self.assertEqual(TimedScheduleInstance.objects.using(self.db2).count(), 0)
+
+        instance = self.make_timed_schedule_instance(self.p2_uuid)
+        save_timed_schedule_instance(instance)
+
+        self.assertEqual(TimedScheduleInstance.objects.using(partition_config.get_proxy_db()).count(), 0)
+        self.assertEqual(TimedScheduleInstance.objects.using(self.db1).count(), 0)
+        self.assertEqual(TimedScheduleInstance.objects.using(self.db2).count(), 1)
+
+    def test_get_alert_schedule_instance(self):
+        self.test_save_alert_schedule_instance()
+        instance = get_alert_schedule_instance(self.p1_uuid)
+        self.assertTrue(isinstance(instance, AlertScheduleInstance))
+        self.assertEqual(instance.schedule_instance_id, self.p1_uuid)
+
+        with self.assertRaises(AlertScheduleInstance.DoesNotExist):
+            get_alert_schedule_instance(uuid.uuid4())
+
+    def test_get_timed_schedule_instance(self):
+        self.test_save_timed_schedule_instance()
+        instance = get_timed_schedule_instance(self.p2_uuid)
+        self.assertTrue(isinstance(instance, TimedScheduleInstance))
+        self.assertEqual(instance.schedule_instance_id, self.p2_uuid)
+
+        with self.assertRaises(TimedScheduleInstance.DoesNotExist):
+            get_timed_schedule_instance(uuid.uuid4())
+
+    def test_access_restricted(self):
```

Comment:
I'm referring specifically to the `test_access_restricted` rests which are identical between the two files and since they just access restriction I don't see the need for them to be repeated for sharded and non-sharded envs.